### ISSUE TITLE
Deeply nested interface data objects rendered incorrectly in UI

### ISSUE THREAD

**Issue by benvinegar** [getsentry/sentry#3145]
First reported here: https://twitter.com/obmarg/status/726060423435407360

![image](https://cloud.githubusercontent.com/assets/2153/14925159/d950680a-0dfa-11e6-806d-ae34dc899669.png)

**Comment by mitsuhiko** [comment]
It basically happens because of lots of repr() because of stripping when we detect too deep nesting. Not entirely sure what the best way to fix this is.

**Comment by obmarg** [comment]
I'm not sure what the definition of too deep nesting is, but the data in that screenshot is only about 4 levels deep

**Comment by dentarg** [comment]
Seems like this is the same as https://github.com/getsentry/sentry/issues/5303

Anyway, I ran into this too, took me a while to understand where the problem was. Here's an easy [raven-ruby](https://github.com/getsentry/raven-ruby) repro:

```ruby
data = JSON.parse('{ "foo": [ { "bar": { "baz": "åäö" } } ] }')
Raven.extra_context(data: data)
```

**Comment by EvanPurkhiser** [comment]
Duplicate of #5303. Let's track over there.

**Comment by alex-hofsteede** [comment]
This wasn't quite a duplicate of #5303, but related. This particular issue with the escaping was solved in 77655087
